### PR TITLE
Fetch: Access-Control-Allow-Method/Header: *

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/fetch/issues/251.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5050)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208258327)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 081238637d3c4e3be78881c1dc77b1ad249b49f9
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5050/fetch/api/cors/cors-preflight-star-worker.html">/fetch/api/cors/cors-preflight-star-worker.html</a></summary>


|                                                   Subtest                                                    | Results |                                                       Messages                                                      |
|--------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------------------------|
|                                                                                                              | OK      |                                                                                                                     |
| `CORS that succeeds with credentials: false; method: GET (allowed: get); header: X-Test,1 (allowed: x-test)` | PASS    | `{}`                                                                                                                |
| `CORS that succeeds with credentials: false; method: SUPER (allowed: *); header: X-Test,1 (allowed: x-test)` | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: NetworkError when attempting to fetch resource."` |
| `CORS that succeeds with credentials: false; method: OK (allowed: *); header: X-Test,1 (allowed: *)`         | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: NetworkError when attempting to fetch resource."` |
| `CORS that fails with credentials: true; method: OK (allowed: *); header: X-Test,1 (allowed: *)`             | PASS    | `{}`                                                                                                                |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: )`            | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`                                        |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: *)`           | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`                                        |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5050/fetch/api/cors/cors-preflight-star.html">/fetch/api/cors/cors-preflight-star.html</a></summary>


|                                                   Subtest                                                    | Results |                                                       Messages                                                      |
|--------------------------------------------------------------------------------------------------------------|---------|---------------------------------------------------------------------------------------------------------------------|
|                                                                                                              | OK      |                                                                                                                     |
| `CORS that succeeds with credentials: false; method: GET (allowed: get); header: X-Test,1 (allowed: x-test)` | PASS    |                                                                                                                     |
| `CORS that succeeds with credentials: false; method: SUPER (allowed: *); header: X-Test,1 (allowed: x-test)` | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: NetworkError when attempting to fetch resource."` |
| `CORS that succeeds with credentials: false; method: OK (allowed: *); header: X-Test,1 (allowed: *)`         | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: NetworkError when attempting to fetch resource."` |
| `CORS that fails with credentials: true; method: OK (allowed: *); header: X-Test,1 (allowed: *)`             | PASS    |                                                                                                                     |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: )`            | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`                                        |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: *)`           | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`                                        |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208258328)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 081238637d3c4e3be78881c1dc77b1ad249b49f9
Using browser at version 58.0.3026.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5050/fetch/api/cors/cors-preflight-star-worker.html">/fetch/api/cors/cors-preflight-star-worker.html</a></summary>


|                                                   Subtest                                                    | Results |                                       Messages                                      |
|--------------------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------|
|                                                                                                              | OK      |                                                                                     |
| `CORS that succeeds with credentials: false; method: GET (allowed: get); header: X-Test,1 (allowed: x-test)` | PASS    | `{}`                                                                                |
| `CORS that succeeds with credentials: false; method: SUPER (allowed: *); header: X-Test,1 (allowed: x-test)` | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: Failed to fetch"` |
| `CORS that succeeds with credentials: false; method: OK (allowed: *); header: X-Test,1 (allowed: *)`         | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: Failed to fetch"` |
| `CORS that fails with credentials: true; method: OK (allowed: *); header: X-Test,1 (allowed: *)`             | PASS    | `{}`                                                                                |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: )`            | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`        |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: *)`           | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`        |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5050/fetch/api/cors/cors-preflight-star.html">/fetch/api/cors/cors-preflight-star.html</a></summary>


|                                                   Subtest                                                    | Results |                                       Messages                                      |
|--------------------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------|
|                                                                                                              | OK      |                                                                                     |
| `CORS that succeeds with credentials: false; method: GET (allowed: get); header: X-Test,1 (allowed: x-test)` | PASS    |                                                                                     |
| `CORS that succeeds with credentials: false; method: SUPER (allowed: *); header: X-Test,1 (allowed: x-test)` | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: Failed to fetch"` |
| `CORS that succeeds with credentials: false; method: OK (allowed: *); header: X-Test,1 (allowed: *)`         | FAIL    | `promise_test: Unhandled rejection with value: object "TypeError: Failed to fetch"` |
| `CORS that fails with credentials: true; method: OK (allowed: *); header: X-Test,1 (allowed: *)`             | PASS    |                                                                                     |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: )`            | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`        |
| `CORS that fails with credentials: true; method: GET (allowed: *); header: undefined (allowed: *)`           | FAIL    | `assert_unreached: Should have rejected: undefined Reached unreachable code`        |

</details>
</details>

File: fetch/api/cors/cors-preflight-star.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+if (this.document === undefined) {
+  importScripts("/resources/testharness.js");
+  importScripts("../resources/utils.js");
+  importScripts("/common/get-host-info.sub.js");
+}
+
+const url = get_host_info().HTTP_REMOTE_ORIGIN + dirname(location.pathname) + RESOURCES_DIR + "preflight.py",
+      origin = location.origin // assuming an ASCII origin
+
+
```
Comment (youennf) [inline fetch/api/cors/cors-preflight-star.js]: two lines there.

**Review by youennf — APPROVED** [review]

**Comment by annevk** [comment]
Updated existing browser bugs with links to these tests (see Fetch issue for pointers).

**Pull request merged**